tests/test_circuit.py

```diff
@@ -78,3 +77,10 @@ def _fwd(phi: torch.Tensor, theta: torch.Tensor, epsilon: torch.Tensor) -> torch
         return circ(state, {"phi": phi, "theta": theta, "epsilon": epsilon})
 
     assert torch.autograd.gradcheck(_fwd, (phi, theta, epsilon))
+
+
+def test_device_inference() -> None:
+    ops = [pyq.RX(0), pyq.RX(0)]
+    circ = pyq.QuantumCircuit(2, ops)
+    nested_circ = pyq.QuantumCircuit(2, [circ, circ])
+    assert nested_circ._device is not None
```

Comment:
I guess this is then a case that failed in the past?

Reply:
no but i just wanted to have a test in the suite which consistently checks that this property correctly works